### PR TITLE
Bump spd logto1.3.1

### LINKED ISSUE #188: Update shipped version of spdlog
**Is your feature request related to a problem? Please describe.**
AutoPas currently ships with spdlog version 0.16.3, which was released on 12 Jan 2018. On 13 Jan 2019, version 1.3.1 was released, which contains various bug fixes and enhancements.

**Describe the solution you'd like**
Update shipped spdlog to version 1.3.1+ to avoid problems, which might occur if we switch to newer compilers.

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

Updates the bundled version of spdlog to 1.3.1.

## TODOs
- [x] ~Replace verbose logging macro with new formatting functionality~ too much work to be worth it
- [x] colors?
- [x] examine what other cool features might be useful
- [x] clean up bundled version

## Resolved Issues # (issue)

- [x] closes #188 

# How Has This Been Tested?

- [x] Jenkins

### CODE REVIEW

**Comment by Garfield96** [comment]
CMake option `COLORED_LOGGING` (default: OFF) enables basic coloring for console output:
Logging_with_coloring
Log messages printed to other destinations than `std::cout` and `std::cerr` are printed without coloring to avoid messed up log files:
Logging_without_coloring
`COLORED_LOGGING` and `VERBOSE_LOGGING` can be used independently.

File: src/autopas/utils/Logger.h [hunk 1]
```diff
@@ -80,7 +73,18 @@ class Logger {
...
     // drop an already registered Logger if it exists
     if (spdlog::get(loggerName())) spdlog::drop(loggerName());
-    auto ostream_sink = std::make_shared<spdlog::sinks::ostream_sink_mt>(oss);
+    std::shared_ptr<spdlog::sinks::sink> ostream_sink;
+#ifdef COLORED_CONSOLE_LOGGING
+    if (oss.rdbuf() == std::cout.rdbuf()) {
+      ostream_sink = std::make_shared<spdlog::sinks::stdout_color_sink_mt>();
+    } else if (oss.rdbuf() == std::cerr.rdbuf()) {
+      ostream_sink = std::make_shared<spdlog::sinks::stderr_color_sink_mt>();
+    } else {
```
Comment (FG-TUM) [inline src/autopas/utils/Logger.h]:
```suggestion
    } else { // no color for streams other than cout / cerr
```

**Comment by FG-TUM** [comment]
@SteffenSeckler review plz. I can't since I'm the creator of this PR ;)

File: src/autopas/utils/Logger.h [hunk 2]
```diff
@@ -80,7 +73,18 @@ class Logger {
   static void create(std::ostream& oss = std::cout) {
     // drop an already registered Logger if it exists
     if (spdlog::get(loggerName())) spdlog::drop(loggerName());
-    auto ostream_sink = std::make_shared<spdlog::sinks::ostream_sink_mt>(oss);
+    std::shared_ptr<spdlog::sinks::sink> ostream_sink;
+#ifdef COLORED_CONSOLE_LOGGING
```
Comment (SteffenSeckler) [inline src/autopas/utils/Logger.h]: should get an AUTOPAS prefix

Comment (SteffenSeckler) [reply src/autopas/utils/Logger.h]: i.e. AUTOPAS_COLORED_CONSOLE_LOGGING

File: cmake/modules/logging.cmake [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
-target_compile_definitions(autopas
-        PUBLIC
-        AUTOPAS_VERBOSE_LOG
-        )
-
-message(STATUS "Verbose log messages enabled.")
+if (COLORED_LOGGING)
+    target_compile_definitions(autopas
+            PUBLIC
+            COLORED_CONSOLE_LOGGING
```
Comment (SteffenSeckler) [inline cmake/modules/logging.cmake]: same as below: should be AUTOPAS_COLORED_CONSOLE_LOGGING

**Comment by SteffenSeckler** [comment]
needs clang format

**Comment by SteffenSeckler** [comment]
btw. the error on lgtm is caused by this:
```
[2019-05-31 16:16:38] [build] Get:1 http://archive.ubuntu.com/ubuntu cosmic/universe amd64 libspdlog-dev amd64 1:0.17.0-1 [149 kB]
```
so it installs an old version of spdlog...

probably it is best to always use the bundled version, i.e., remove the line
```
find_package(spdlog CONFIG QUIET)
```

**Comment by Garfield96** [comment]
lgtm alert is caused by fmt (bundled with spdlog) and can be ignored. I created an issue for this alert inside the fmt repo.

**Pull request merged**